internal/impl/devbox.go

```diff
@@ -724,13 +725,31 @@ func (d *Devbox) computeNixEnv(setFullPath bool) (map[string]string, error) {
 		}
 	}
 
-	// Overwrite/leak whitelisted vars into nixEnv:
-	for name, leak := range leakedVars {
-		if leak {
-			env[name] = os.Getenv(name)
+	// Copy over (and overwrite) vars that we explicitly "leak", as well as DEVBOX_ vars.
+	for _, kv := range os.Environ() {
+		parts := strings.SplitN(kv, "=", 2)
+		key := parts[0]
+		val := parts[1]
+
+		if strings.HasPrefix(parts[0], "DEVBOX_") {
+			env[key] = val
+		}
+
+		if _, ok := leakedVars[key]; ok {
+			env[key] = val
+		}
+
+		if _, ok := leakedVarsForShell[key]; ok {
+			env[key] = val
 		}
 	}
 
+	// These variables are only needed for shell, but we include them here in the computed env
+	// for both shell and run in order to be as identical as possible.
+	env["NIX_PKGS_ALLOW_UNFREE"] = "1"     // Only for shell because we don't expect nix calls within run
```

Comment:
I think its good to include, because someone's script may do `devbox add` for them.